src/main/java/org/deepsymmetry/beatlink/VirtualRekordbox.java

```diff
@@ -368,13 +362,45 @@ private DeviceUpdate buildUpdate(DatagramPacket packet) {
                 byte[] data = packet.getData();
                 // PSSI Data
                 if (data[0x25] == 10) {
+
                     int rekordboxId = (int) Util.bytesToNumber(data, 0x28, 4);
-                    // If track is loaded
-                    if (rekordboxId != 0) {
+                    // If track is loaded and OpusProvider has attached media
+                    if (rekordboxId != 0 && OpusProvider.getInstance().hasAttachedArchive()) {
                         final byte[] pssiFromOpus = Arrays.copyOfRange(data, 0x35, data.length);
-                        final CdjStatus.TrackSourceSlot slot = (data[0x2d] == 3) ? SD_SLOT: USB_SLOT;
-                        final int player = data[0x21];
-                        OpusProvider.getInstance().handlePSSIMatching(rekordboxId, pssiFromOpus, player, slot);
+                        final int player = Util.translateOpusPlayerNumbers(data[0x21]);
+
+                        OpusProvider.getInstance().handlePSSIMatching(rekordboxId, pssiFromOpus, player);
+
+                        SlotReference slotRef = SlotReference.getSlotReference(player, USB_SLOT);
+
+                        if (MetadataFinder.getInstance().getMediaDetailsFor(slotRef) == null) {
```

Comment:
This is wrong: no media update is sent for an empty slot. But all this code is going away anyway, and will be replaced by code that creates the details and sends it as an update when the user attaches a metadata archive.